pkg/volume/iscsi/iscsi_util.go

```diff
@@ -695,31 +733,33 @@ func (util *ISCSIUtil) DetachBlockISCSIDisk(c iscsiDiskUnmapper, mapPath string)
 		if err != nil {
 			return err
 		}
-		arr := strings.Split(device, "-lun-")
-		if len(arr) < 2 {
-			return fmt.Errorf("failed to retrieve lun from mapPath: %v", mapPath)
-		}
-		lun = arr[1]
 		// Extract the iface from the mountPath and use it to log out. If the iface
 		// is not found, maintain the previous behavior to facilitate kubelet upgrade.
 		// Logout may fail as no session may exist for the portal/IQN on the specified interface.
 		iface, found = extractIface(mapPath)
 	}
+
+	// Delete all the scsi devices and any multipath devices
+	if err = deleteDevices(c.iscsiDisk, c.exec, c.deviceUtil); err != nil {
```

Comment:
the block detach path did not call this `deleteDevices` before. Why? I am pretty sure it should.

Reply:
After this change and the below removal of os.Stat(devicePath), `DetachBlockISCSIDisk` and `DetachDisk` will look almost identical except `DetachDisk` tries to unmount the mntPath